boot/boot_serial/src/boot_serial.c

```diff
@@ -186,17 +187,34 @@ bs_list(char *buf, int len)
             if (flash_area_open(area_id, &fap)) {
                 continue;
             }
+#ifdef MCUBOOT_IMAGE_ACCESS_HOOKS
+            int rc = boot_read_image_header_hook(image_index, slot, &hdr);
+            if (rc == BOOT_HOOK_REGULAR)
+#endif
+            {
+                flash_area_read(fap, 0, &hdr, sizeof(hdr));
+            }
 
-            flash_area_read(fap, 0, &hdr, sizeof(hdr));
+            fih_int fih_rc = FIH_FAILURE;
 
-            if (hdr.ih_magic != IMAGE_MAGIC ||
-              bootutil_img_validate(NULL, 0, &hdr, fap, tmpbuf, sizeof(tmpbuf),
-                                    NULL, 0, NULL)) {
-                flash_area_close(fap);
-                continue;
+            if (hdr.ih_magic == IMAGE_MAGIC)
+            {
+#ifdef MCUBOOT_IMAGE_ACCESS_HOOKS
+                FIH_CALL(boot_image_check_hook, fih_rc, image_index, slot);
+                if (fih_eq(fih_rc, BOOT_HOOK_REGULAR))
+#endif
```

Comment:
I think having ifdefs that don't match up with the C structure bother a lot more than having all of the ifdefs. Can we at least put this ifdef in a header, and define these as empty macros (well, ones like @de-nordic showed that don't generate warnings) when the feature isn't defined. That would at least move the clutter to a header instead of all through out the source code.